File: vuln-sandbox/node/cwe-798/aws-hardcoded-access-key-safe.js
// SAFE: aws-hardcoded-access-key — credentials loaded from environment variables, not hardcoded
// Rule: AwsHardcodedAccessKey | CWE-798 | Expected: TrueNegative

const { S3Client, ListBucketsCommand } = require('@aws-sdk/client-s3');

// SAFE: credentials loaded from environment variables or IAM role, never hardcoded
const client = new S3Client({
  region: process.env.AWS_REGION || 'us-east-1',
  // No credentials specified — SDK uses environment variables or instance profile automatically
});

async function listBuckets() {
  const command = new ListBucketsCommand({});
  const response = await client.send(command);
  return response.Buckets;
}

listBuckets().then(console.log).catch(console.error);

File: vuln-sandbox/python/cwe-798/aws-hardcoded-access-key-safe.py
# SAFE: aws-hardcoded-access-key — credentials loaded from environment or IAM role
# Rule: AwsHardcodedAccessKey | CWE-798 | Expected: TrueNegative

import boto3

# SAFE: no credentials passed explicitly; boto3 uses environment variables,
# ~/.aws/credentials, or the EC2/ECS instance profile automatically
s3 = boto3.client('s3')

def list_buckets():
    response = s3.list_buckets()
    return [b['Name'] for b in response.get('Buckets', [])]

if __name__ == '__main__':
    print(list_buckets())
